.travis.yml

```diff
@@ -59,9 +66,12 @@ before_install:
   - docker exec pmm-managed-server ls /root/.cache/go-build
 
 install:
+  # install devcontainer
   - make env-devcontainer
 
 before_script:
+  # static analyze
+  - make env TARGET=ci-reviewdog
```

Comment:
Not sure if we want to run linters before tests. Why not move it into `travis` target?

Reply:
It's better to get linters feedback earlier to fix ASAP

Reply:
Well, yeah, sure. Just run them locally instead of using CI for that. We may add a separate target for running `golangci-lint --new-from-rev=PMM-2.0` if that would help. But I don't think we should encourage "push and wait" development.

Reply:
idk why but it doesn't work in devcontainer, it returns all errors, not only diff.

Reply:
```
[root@pmm-managed-server pmm-managed]# golangci-lint run | wc
   1571    7334   84166
[root@pmm-managed-server pmm-managed]# golangci-lint run --new-from-rev=PMM-2.0 | wc
     12      68     627
```

Reply:
Any reason to put lint after tests? it's often case when developers push changes before checking in linter and I think developer should see issues earlier then later.

Reply:
> it's often case when developers push changes before checking in linter

Please don't do that. Please do not encourage others to do that. Please do run tests and linters locally.

I'm really worried about the decline of our engineering practices. Somehow we started to think that pushing code to CI and waiting is okay. That pushing code without tests is okay. That tests somehow are not in the estimates. Why?

Reply:
I don't think that it's okay, but we have this problem and it's better to catch the problem earlier. And it's clear that it doesn't work this way just telling people it again and again.

About test related to Alerting rules, I didn't know how to test it as part of API tests and I'm still not sure that it can be tested by checking AlertManager API. I'll check Prometheus API to check if it's updated or not.

Reply:
>  it returns all errors, not only diff.

Mmm, I don't know, but how about fixing all linter warnings? There a lot of redundant `nolint` annotations and different small warnings that easy to fix. We can do it not at once, but each time that we create new PR, to include also a few old warnings fixes.

Reply:
@artemgavrilov it's already fixed.

Reply:
> I'm really worried about the decline of our engineering practices. Somehow we started to think that pushing code to CI and waiting is okay. That pushing code without tests is okay. That tests somehow are not in the estimates. Why?

It sounds like two different problem al-a "client-side vs server-side validation". 
And could be solved as a separate task with for example https://pre-commit.com.

Reply:
> About test related to Alerting rules, I didn't know how to test it as part of API tests and I'm still not sure that it can be tested by checking AlertManager API.

https://github.com/Percona-Lab/pmm-api-tests/pull/94 – that PR, reviewed by you, uses Alertmanager API for almost the same problem. Generate alerting rule, save it, wait for it to appear on Alertmanager.

Reply:
You mentioned two or more times that these are not alermanager rules. So it should be checked as prometheus alerting rules.

Reply:
Please check Slack